### PR TITLE
Refactor spdm_get_measurement_summary_hash to spdm_device_secret_lib.

### PR DESCRIPTION
**Pull request by rkongintel**
Fix for issue:
https://github.com/DMTF/libspdm/issues/267

Signed-off-by: Kong, Richard <richard.kong@intel.com>

### CODE REVIEW

File: library/spdm_responder_lib/libspdm_rsp_challenge_auth.c [hunk 1]
```diff
@@ -171,8 +171,22 @@ return_status spdm_get_response_challenge_auth(IN void *context,
     spdm_get_random_number(SPDM_NONCE_SIZE, ptr);
     ptr += SPDM_NONCE_SIZE;
 
-    result = spdm_generate_measurement_summary_hash(
-        spdm_context, FALSE, spdm_request->header.param2, ptr);
+    if (spdm_is_capabilities_flag_supported(
```
Comment (rkongintel) [inline library/spdm_responder_lib/libspdm_rsp_challenge_auth.c:174]: This capability check was moved out from within the spdm_generate_measurement_summary_hash in order to eliminate the need to pass the spdm_context_t.

File: library/spdm_responder_lib/libspdm_rsp_key_exchange.c [hunk 2]
```diff
@@ -264,8 +264,22 @@ return_status spdm_get_response_key_exchange(IN void *context,
 
     ptr += dhe_key_size;
 
-    result = spdm_generate_measurement_summary_hash(
-        spdm_context, FALSE, spdm_request->header.param1, ptr);
+    if (spdm_is_capabilities_flag_supported(
```
Comment (rkongintel) [inline library/spdm_responder_lib/libspdm_rsp_key_exchange.c:267]: This capability check was moved out from within the spdm_generate_measurement_summary_hash in order to eliminate the need to pass the spdm_context_t.

File: library/spdm_responder_lib/libspdm_rsp_psk_exchange.c [hunk 3]
```diff
@@ -242,8 +242,22 @@ return_status spdm_get_response_psk_exchange(IN void *context,
 
     ptr = (void *)(spdm_response + 1);
 
-    result = spdm_generate_measurement_summary_hash(
-        spdm_context, FALSE, spdm_request->header.param1, ptr);
+    if (spdm_is_capabilities_flag_supported(
```
Comment (rkongintel) [inline library/spdm_responder_lib/libspdm_rsp_psk_exchange.c:245]: This capability check was moved out from within the spdm_generate_measurement_summary_hash in order to eliminate the need to pass the spdm_context_t.

**Comment by jyao1** [comment]
looks good to me.

**Review by jyao1 — APPROVED** [review]

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**